src/ai_chunking.py

```
@@ -0,0 +1,222 @@
+        total_input_tokens += current_chunk_token_count
+
+    logging.debug(f"Number of chunks: {len(message_chunks)}")
+    logging.debug(f"Number of messages: {len(messages)}")
+    logging.debug(f"Total tokens: {total_input_tokens}")
+
+    return message_chunks, messages_per_chunk, chunk_start_messages, total_input_tokens
+
+
+
+# This is an Arjun's implementation of chunking messages for OpenAI API.
+def chunk_messages_by_model_token_limit(messages, model: dict):
```

Ruff (0.8.2)

Redefinition of unused `chunk_messages_by_model_token_limit` from line 55

(F811)